moon-store-mac-os/moon-store-mac-os/Services/BaseNetworkService.swift

```diff
@@ -0,0 +1,157 @@
+            return try await request(urlRequest: urlRequest)
+        } catch {
+            throw NSError(domain: "Error posting data: \(error.localizedDescription)", code: 500)
+        }
+        
+        
+    }
+    
+    // MARK: - HTTP POST Multipart
+    
+    func postMultipartData(for path: MSPath,
+                           with parameters: [String: Any],
+                           dataSet: [Data]) async throws -> Data {
+        components.path = path.endpoint
+        components.queryItems = makeQueryItems(parameters: parameters)
+        
+        guard let url = components.url else {
+            throw MSError.badUrl
+        }
+        
+        let boundary = "Boundary-\(UUID().uuidString)"
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = Constants.POST
+        urlRequest.setValue("multipart/form-data; boundary=\(boundary)", forHTTPHeaderField: "Content-Type")
+        
+        var data = Data()
+        
+        for (key, value) in parameters {
+            data.appendString("--\(boundary)\r\n")
+            data.appendString("Content-Disposition: form-data; name=\"\(key)\"\r\n\r\n")
+            data.appendString("\(value)\r\n")
+        }
+        
+        for value in dataSet {
+            data.appendString("--\(boundary)\r\n")
+            data.appendString("Content-Disposition: form-data; name=\"images\"; filename=\"image-\(boundary).jpg\"\r\n")
+            data.appendString("Content-Type: image/jpeg\r\n\r\n")
+            data.append(value)
+            data.appendString("\r\n")
+        }
+        
+        data.appendString("--\(boundary)--\r\n")
+        
+        urlRequest.httpBody = data
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    // MARK: - HTTP DELETE
+    
+    func deleteData(for path: MSPath, with parameters: [String: Any]) async throws -> Data {
+        components.path = path.endpoint
+        components.queryItems = makeQueryItems(parameters: parameters)
+        
+        guard let url = components.url else {
+            throw MSError.badUrl
+        }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = Constants.DELETE
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    // MARK: - HTTP UPDATE
+    
+    func putData(for path: MSPath, with parameters: [String: Any]) async throws -> Data {
+        components.path = path.endpoint
+        components.queryItems = makeQueryItems(parameters: parameters)
+        
+        guard let url = components.url else {
+            throw MSError.badUrl
+        }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = Constants.PUT
+        
+        return try await request(urlRequest: urlRequest)
+    }
+    
+    func request(urlRequest: URLRequest) async throws -> Data {
+        do {
+            let (data, _) = try await URLSession.shared.data(for: urlRequest)
```

Comment:
handle response or remove it

Reply:
It will be handle on another ticket pertaining to error handler. We need to add more logic.